src/lib/homestore_backend/replication_state_machine.cpp

```diff
@@ -46,8 +46,50 @@ void ReplicationStateMachine::on_rollback(int64_t lsn, sisl::blob const&, sisl::
 
 void ReplicationStateMachine::on_error(ReplServiceError error, const sisl::blob& header, const sisl::blob& key,
                                        cintrusive< repl_req_ctx >& ctx) {
-    LOGE("on_error  with :{}, lsn {}", error, ctx->lsn);
-    // TODO:: block is already freeed at homestore side, handle error if necessay.
+
+    RELEASE_ASSERT(ctx->is_proposer, "on_error should only be called from proposer");
+    const ReplicationMessageHeader* msg_header = r_cast< const ReplicationMessageHeader* >(header.cbytes());
+    LOGE("on_error, message type {} with lsn {}, error {}", msg_header->msg_type, ctx->lsn, error);
+    switch (msg_header->msg_type) {
+    case ReplicationMessageType::CREATE_PG_MSG: {
+        if (ctx) {
```

Comment:
we do not need `if (ctx)` here, since on_error will never be call in recovery mode and will always be called in leader.
on the other hand,  before `ctx->is_proposer` , we should make sure ctx is not nullptr

so , remove all the `if(ctx)` here and add a line before line 50:
`RELEASE_ASSERT(ctx , "ctx should not be nullptr in on_error");`

Reply:
thanks, addressed